java/core/src/main/java/com/microsoft/bond/Bonded.java

```diff
@@ -135,6 +135,19 @@
      */
     public abstract <U extends BondSerializable> Bonded<U> convert(StructBondType<U> toBondType);
 
+    public <U extends BondSerializable> Bonded<U> cast(StructBondType<U> castBondType) {
```

Comment:
Can we document this method? The documentation should contrast `cast` with `convert`, and the documentation for `convert` should probably talk about `cast` as well.